web/tests/api-spec.js

```diff
@@ -4,13 +4,15 @@ describe('api service', function () {
         fixture.setBase('tests/api-description');
 
         inject(function ($http) {
-            const MOCK_PROJECT_SETTINGS = {
-                API_ROOT: 'localhost:8000'
+            const mockProjectConfig = {
+                getSettings:() => ({
+                    apiRoot: 'localhost:8000'
+                })
```

Comment:
Hmm that's a strong position. I can see arrow funcitons playing nice in promises:

`.then((response) => request.data)`
Or in a `d3` context `.attr('x', (data) => data.height)`